src/CMakeLists.txt

```diff
@@ -58,6 +58,7 @@ endif()
 
 # Compiler `-D*` definitions
 add_definitions(-DNTA_PLATFORM_${PLATFORM}${BITNESS}
+                -DNTA_ARCH_${BITNESS}
```

Comment:
How about add also NTA_ARCH_DARWIN, NTA_ARCH_LINUX and NTA_ARCH_WINDOWS here:

https://github.com/numenta/nupic.core/blob/master/src/CMakeLists.txt#L40

```
if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
  set(PLATFORM "darwin")
  set(STDLIB "-stdlib=libc++")
  add_definitions(-DNTA_OS_DARWIN)
elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
  set(PLATFORM "windows")
  set(STDLIB "")
  add_definitions(-DNTA_OS_WINDOWS)
elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
  set(PLATFORM "linux")
  set(STDLIB "")
  add_definitions(-DNTA_OS_LINUX)
endif()
```